boot/bootutil/include/bootutil/image.h

```diff
@@ -28,34 +28,38 @@ extern "C" {
 
 struct flash_area;
 
-#define IMAGE_MAGIC                 0x96f3b83c
+#define IMAGE_MAGIC                 0x96f3b83d
+#define IMAGE_MAGIC_V1              0x96f3b83c
+#define IMAGE_TRAILER_MAGIC         0x6907
 #define IMAGE_MAGIC_NONE            0xffffffff
 
 #define IMAGE_HEADER_SIZE           32
 
 /*
  * Image header flags.
  */
-#define IMAGE_F_PIC                      0x00000001 /* Not supported. */
-#define IMAGE_F_SHA256                   0x00000002 /* Hash TLV is present */
-#define IMAGE_F_PKCS15_RSA2048_SHA256    0x00000004 /* PKCS15 w/RSA and SHA */
-#define IMAGE_F_ECDSA224_SHA256          0x00000008 /* ECDSA224 over SHA256 */
-#define IMAGE_F_NON_BOOTABLE             0x00000010 /* Split image app. */
-#define IMAGE_F_ECDSA256_SHA256          0x00000020 /* ECDSA256 over SHA256 */
-#define IMAGE_F_PKCS1_PSS_RSA2048_SHA256 0x00000040 /* PKCS1 PSS */
-
-/*
- * ECSDA224 is with NIST P-224
- * ECSDA256 is with NIST P-256
- */
+#define IMAGE_F_PIC                   0x00000001 /* Not currently supported. */
+#define IMAGE_F_NON_BOOTABLE          0x00000002 /* Split image app. */
 
 /*
  * Image trailer TLV types.
+ *
+ * Signature is generated by computing signature over the image hash.
+ * Currently the only image hash type is SHA256.
+ *
+ * Signature comes in form of 2 TLVs.
+ *   1st one identifies the public key which should be used to verify it.
+ *   2nd one is the actual signature.
  */
-#define IMAGE_TLV_SHA256            1	/* SHA256 of image hdr and body */
-#define IMAGE_TLV_RSA2048           2	/* RSA2048 of hash output */
-#define IMAGE_TLV_ECDSA224          3   /* ECDSA of hash output */
-#define IMAGE_TLV_ECDSA256          4   /* ECDSA of hash output */
+#define IMAGE_TLV_KEYHASH         0x01  /* hash of the public key */
+
+#define IMAGE_TLV_SHA256          0x10	/* SHA256 of image hdr and body */
+
+#define IMAGE_TLV_RSA2048         0x20	/* RSA2048 of hash output */
```

Comment:
As per discussion, I was going to remove the pcks1v15 version and just keep the PSS.
I'll include PSS in the define name.